src/components/Chat/ChatContent/ChatContent.tsx

```diff
@@ -14,107 +14,106 @@ import CloneChat from './CloneChat';
-      ? state.chats[state.currentChatIndex].messages
-      : []
-  );
-  const stickyIndex = useStore((state) =>
-    state.chats &&
-    state.chats.length > 0 &&
-    state.currentChatIndex >= 0 &&
-    state.currentChatIndex < state.chats.length
-      ? state.chats[state.currentChatIndex].messages.length
-      : 0
-  );
-  const advancedMode = useStore((state) => state.advancedMode);
-  const generating = useStore.getState().generating;
-  const hideSideMenu = useStore((state) => state.hideSideMenu);
+   const inputRole = useStore((state) => state.inputRole);
+   const setError = useStore((state) => state.setError);
+   const messages = useStore((state) =>
+      state.chats &&
+         state.chats.length > 0 &&
+         state.currentChatIndex >= 0 &&
+         state.currentChatIndex < state.chats.length
+         ? state.chats[state.currentChatIndex].messages
+         : []
+   );
+   const stickyIndex = useStore((state) =>
+      state.chats &&
+         state.chats.length > 0 &&
+         state.currentChatIndex >= 0 &&
+         state.currentChatIndex < state.chats.length
+         ? state.chats[state.currentChatIndex].messages.length
+         : 0
+   );
+   const advancedMode = useStore((state) => state.advancedMode);
+   const generating = useStore.getState().generating;
+   const hideSideMenu = useStore((state) => state.hideSideMenu);
 
-  const saveRef = useRef<HTMLDivElement>(null);
+   const saveRef = useRef<HTMLDivElement>(null);
 
-  // clear error at the start of generating new messages
-  useEffect(() => {
-    if (generating) {
-      setError('');
-    }
-  }, [generating]);
+   // clear error at the start of generating new messages
+   useEffect(() => {
+      if (generating) {
+         setError('');
+      }
+   }, [generating]);
 
-  const { error } = useSubmit();
+   const { error } = useSubmit();
 
-  return (
-    <div className='flex-1 overflow-hidden'>
-      <ScrollToBottom
-        className='h-full dark:bg-gray-800'
-        followButtonClassName='hidden'
-      >
-        <ScrollToBottomButton />
-        <div className='flex flex-col items-center text-sm dark:bg-gray-800'>
-          <div
-            className='flex flex-col items-center text-sm dark:bg-gray-800 w-full'
-            ref={saveRef}
-          >
-            <ModelConfigBar />
-            {!generating && advancedMode && messages?.length === 0 && (
-              <NewMessageButton messageIndex={-1} />
-            )}
-            {messages?.map((message, index) => (
-              <React.Fragment key={index}>
-                <Message
-                  role={message.role}
-                  content={message.content}
-                  messageIndex={index}
-                />
-                {!generating && advancedMode && <NewMessageButton messageIndex={index} />}
-              </React.Fragment>
-            ))}
-          </div>
+   return (
+      <div className='flex-1 scroll-smooth overflow-hidden'>
```

Comment:
smooth scrolling - helps scolling not be so jarring (I'd like an option to turn auto-scrolling off eventually!)